modules/logging-loki-storage.adoc

```diff
@@ -0,0 +1,249 @@
+----
+
+** Suggested secret name: `logging-loki-s3`
+
+** Secret type: `s3`
+
+== Azure
+
+.Prerequisites
+* Loki Operator deployed
+* Create a link:https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction[bucket] on Azure.
+
+.Procedure
+* Create an Object Storage secret with keys as follows:
+
+[source,terminal]
+----
+oc create secret generic logging-loki-azure \
+  --from-literal=container="<AZURE_CONTAINER_NAME>" \
+  --from-literal=environment="<AZURE_ENVIRONMENT>" \
+  --from-literal=account_name="<AZURE_ACCOUNT_NAME>" \
+  --from-literal=account_key="<AZURE_ACCOUNT_KEY>"
+----
+
+** Suggested secret name: `logging-loki-azure`
+
+** Secret type: `azure`
+
+
+== Google Cloud Storage
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* Create a link:https://cloud.google.com/resource-manager/docs/creating-managing-projects[project] on Google Cloud Platform.
+
+* Create a link:https://cloud.google.com/storage/docs/creating-buckets[bucket] under same project.
+
+* Create a link:https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account[service account] under same project for GCP authentication.
+
+.Procedure
+
+. Copy the service account credentials received from GCP into a file name `key.json`.
+
+. Create an Object Storage secret with keys `bucketname` and `key.json` as follows:
+
++
+[source,terminal]
+----
+oc create secret generic logging-loki-gcs \
+  --from-literal=bucketname="<BUCKET_NAME>" \
+  --from-file=key.json="<PATH/TO/KEY.JSON>"
+----
+
+Where `logging-loki-gcs` is the secret name, `<BUCKET_NAME>` is the name of bucket created in prerequisites step and `<PATH/TO/KEY.JSON>` is the file path where the `key.json` was copied to.
+
+** Suggested secret name: `logging-loki-gcs`
+
+** Secret type: `gcs`
+
+
+== Minio
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* link:https://operator.min.io/[Minio] deployed on your Cluster.
+
+* Create a link:https://docs.min.io/docs/minio-client-complete-guide.html[bucket] on Minio via CLI.
+
+.Procedure
+
+* Create an Object Storage secret with keys as follows:
+
+[source,terminal]
+----
+oc create secret generic logging-loki-minio \
+  --from-literal=bucketnames="<BUCKET_NAME>" \
+  --from-literal=endpoint="<MINIO_BUCKET_ENDPOINT>" \
+  --from-literal=access_key_id="<MINIO_ACCESS_KEY_ID>" \
+  --from-literal=access_key_secret="<MINIO_ACCESS_KEY_SECRET>"
+----
+
+** Suggested secret name: `logging-loki-minio`
+
+** Secret type: `s3`
+
+
+== OpenShift Data Foundation
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* Deploy the [OpenShift Data Foundation](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.12) on your cluster.
+
+.Procedure
+
+. Create an ObjectBucketClaim in `openshift-logging` namespace:
++
+[source,yaml]
+----
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: loki-bucket-odf
+  namespace: openshift-logging
+spec:
+  generateBucketName: loki-bucket-odf
```

Comment:
When I try to create ObjectBucketClaim with above yaml, I get below error:
```
$ cat << EOF | oc create -f -
> apiVersion: objectbucket.io/v1alpha1
> kind: ObjectBucketClaim
> metadata:
>   name: loki-bucket-odf
>   namespace: openshift-logging
> spec:
>   generateBucketName: loki-bucket-odf
> EOF
The ObjectBucketClaim "loki-bucket-odf" is invalid: spec.storageClassName: Required value
```

Reply:
@periklis -- Can you assist with the above please?

Reply:
@periklis ?

Reply:
I can confirm that `storageClassName` is a required field for this resource. I have [found an example](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.13/html/managing_hybrid_and_multicloud_resources/object-bucket-claim#dynamic-object-bucket-claim_rhodf) that uses `openshift-storage.noobaa.io` which looks like it might be a "constant value" for ODF. I have not run a test cluster with ODF on it in a while, so I unfortunately don't remember anymore what value I used for my `ObjectBucketClaim`.